### PR TITLE
docs: add description for program stage working lists [DHIS2-14367]

### PR DESCRIPTION
**Pull request by muilpp**

### CODE REVIEW

File: src/developer/web-api/new-tracker.md [hunk 1]
```diff
@@ -336,6 +336,37 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
 
 > One between `uid` or `username` field must be provided. If both are provided, only username is considered.
 
+### Program stage working lists
+
+If you want to make changes to a program stage working list, such as creating, updating, or deleting it, you'll have to utilize the endpoint called *programStageWorkingLists*. These lists are used in the Capture app to present predetermined working lists that are relevant a specific program stage in the tracker user interface.
```
Comment (enricocolasante) [inline src/developer/web-api/new-tracker.md]:
```suggestion
If you want to make changes to a program stage working list, such as creating, updating, or deleting it, you'll have to utilize the endpoint called *programStageWorkingLists*. These lists are used in the Capture app to present predetermined working lists that are relevant to a specific program stage in the tracker user interface.
```

File: src/developer/web-api/new-tracker.md [hunk 2]
```diff
@@ -336,6 +336,37 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+
+
+
```
Comment (enricocolasante) [inline src/developer/web-api/new-tracker.md:350]: We probably do not have a standard to document enpoints and CRUD operations but I wouldn't understand how to call the API just from the description of the payload.
Maybe one example would make it more clear

File: src/developer/web-api/new-tracker.md (hunk 1 above)
Comment (enricocolasante) [inline src/developer/web-api/new-tracker.md]: Don't we also have a GET endpoint for it?

File: src/developer/web-api/new-tracker.md (hunk 2 above)
Comment (muilpp) [reply src/developer/web-api/new-tracker.md:350]: I added another paragraph as an introduction where I explain what is this resource for and what's the endpoint. Do you think this and the payload table would be clear enough for people to understand how to use it? Or would you still add an example?

Comment (teleivo) [reply src/developer/web-api/new-tracker.md:350]: For me API docs like https://stripe.com/docs/api/balance are helpful in understanding the concepts and making the actual requests 😄 To me ideally there is a request I can copy and paste and run against the SL demo DB. Is the feature explained somewhere? If so can we link to it? Right now, I would not know what the feature is for.

We should probably take time to talk about our docs in general, maybe in the design meeting.

File: src/developer/web-api/new-tracker.md [hunk 3]
```diff
@@ -337,6 +337,45 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+Table: Payload
+
+| Payload values | Description | Example |
+|---|---|---|
+| name | Name of the working list. Required. ||
+| description | A description of the working list. ||
+| program | Object containing the id of the program. Required. | { "id" : "uy2gU8kTjF"} |
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]:
```suggestion
| program | Object containing the id of the program. Required. | {"id" : "uy2gU8kTjF"} |
```

File: src/developer/web-api/new-tracker.md [hunk 4]
```diff
@@ -337,6 +337,45 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+| programStage | Object containing the id of the program stage. Required. | { "id" : "oRySG82BKE6"} |
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]:
```suggestion
| programStage | Object containing the id of the program stage. Required. | {"id" : "oRySG82BKE6"} |
```

I think most samples do not have whitespace between `{`. We should stick to one style 😄

File: src/developer/web-api/new-tracker.md [hunk 5]
```diff
@@ -337,6 +337,45 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
 > One between `uid` or `username` field must be provided. If both are provided, only username is considered.
 
 
+### Program stage working lists { #webapi_working_list_filters } 
+
+The program stage working lists are used in the Capture app to present predetermined working lists that are relevant to a specific program stage in the tracker user interface. To interact with them, you'll need to use the */api/trackedEntityInstanceFilters* resource. These lists can be shared and follow the same sharing pattern as any other metadata object. When using the */api/sharing* the type parameter will be *programStageWorkingLists*.
+
+    /api/40/programStageWorkingLists
+
+
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]: I know we don't have a markdown style guide 😅 I would stick to one newline before and after sections and only one between paragraphs.

File: src/developer/web-api/new-tracker.md (hunk 2 above)
Comment (muilpp) [reply src/developer/web-api/new-tracker.md:350]: The docs in the link you provided look great, but we are kind of far away from that at the moment, would be great if that was our goal though. For now, I extended the description of the endpoint (I hope it's clear), and added an example payload,

File: src/developer/web-api/new-tracker.md [hunk 6]
```diff
@@ -336,7 +336,78 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+| Payload values | Description | Example |
+|---|---|---|
+| name | Name of the working list. Required. ||
+| description | A description of the working list. ||
+| program | Object containing the id of the program. Required. | {"id" : "uy2gU8kTjF"} |
+| programStage | Object containing the id of the program stage. Required. | {"id" : "oRySG82BKE6"} |
+| programStageQueryCriteria | An object representing various possible filtering values. See *Program Stage Query Criteria* definition table below.
+
+Table: Program Stage
```
Comment (enricocolasante) [inline src/developer/web-api/new-tracker.md]:
```suggestion
Table: Program Stage Query Criteria
```

**Review by enricocolasante — APPROVED** [review]
A suggestion on the name of the second table

File: src/developer/web-api/new-tracker.md (hunk 6 above)
Comment (teleivo) [reply src/developer/web-api/new-tracker.md]: @enricocolasante I don't think we should expose internal type names. So we can change such names without having to change the docs. I might get this wrong here. We should only expose our HTTP API like query params

File: src/developer/web-api/new-tracker.md [hunk 7]
```diff
@@ -336,7 +336,78 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
 
 > One between `uid` or `username` field must be provided. If both are provided, only username is considered.
 
+### Program stage working lists { #webapi_working_list_filters } 
 
+The program stage working lists feature within the Capture app is designed to display pre-established working lists relevant to a particular program stage in the tracker user interface. This functionality enables users to save standard filters and sorting preferences that are related to program stages, facilitating the organization and management of their workflow. To interact with them, you'll need to use the */api/programStageWorkingLists* resource. These lists can be shared and follow the same sharing pattern as any other metadata object. When using the */api/sharing* the type parameter will be *programStageWorkingLists*.
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]:
```suggestion
The program stage working lists feature within the Capture app is designed to display pre-established working lists relevant to a particular program stage. This functionality enables users to save filters and sorting preferences that are related to program stages, facilitating the organization and management of their workflow. To interact with them, you'll need to use the */api/programStageWorkingLists* resource. These lists can be shared and follow the same sharing pattern as any other metadata. When using the */api/sharing* the type parameter will be *programStageWorkingLists*.
```

Great 😄 I removed a few words that seem redundant to me so the description gets more succinct.

File: src/developer/web-api/new-tracker.md [hunk 8]
```diff
@@ -336,7 +336,78 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+
+    /api/40/programStageWorkingLists
+
+##### Payload on CRUD operations to program stage working lists
+
+If you want to fetch or make changes to a program stage working list, besides the endpoint mentioned above, you'll need to provide a payload in the following format: 
+
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]: One empty line between paragraphs, headers, ... is enough for adding a line between these in the rendered output. I think we should stick to that 😄

File: src/developer/web-api/new-tracker.md [hunk 9]
```diff
@@ -336,7 +336,78 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
 
+### Program stage working lists { #webapi_working_list_filters } 
 
+The program stage working lists feature within the Capture app is designed to display pre-established working lists relevant to a particular program stage in the tracker user interface. This functionality enables users to save standard filters and sorting preferences that are related to program stages, facilitating the organization and management of their workflow. To interact with them, you'll need to use the */api/programStageWorkingLists* resource. These lists can be shared and follow the same sharing pattern as any other metadata object. When using the */api/sharing* the type parameter will be *programStageWorkingLists*.
+
+    /api/40/programStageWorkingLists
+
+##### Payload on CRUD operations to program stage working lists
+
+If you want to fetch or make changes to a program stage working list, besides the endpoint mentioned above, you'll need to provide a payload in the following format:
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md]: Do I need to provide the same payload when I fetch a `program stage working list` vs when I create one? Its not clear to me

File: src/developer/web-api/new-tracker.md [hunk 10]
```diff
@@ -336,7 +336,78 @@ Tracker notes do not have their dedicated endpoint; they are exchanged as part o
...
+
+See an example payload below:
+
+```json
+{   
+    "name":"Test WL",
+    "program":{"id":"uy2gU8kT1jF"},
+    "programStage":{"id":"oRySG82BKE6"},
+    "description": "Test WL definition",
+    "programStageQueryCriteria":
```
Comment (teleivo) [inline src/developer/web-api/new-tracker.md:385]: I did not know `criteria` is part of our API 😬 I was hoping we can settle on a common language for tracker when it comes to the params, search params, criteria, ...

Comment (muilpp) [reply src/developer/web-api/new-tracker.md:385]: We chose this name to follow with what had been done in the program stage filter, which uses `EventQueryCriteria`, but we can try to improve the naming if needed.

File: src/developer/web-api/new-tracker.md (hunk 6 above)
Comment (muilpp) [reply src/developer/web-api/new-tracker.md]: You mean it for program stage query criteria? It's also used in the API call, so the clients need to be aware of it.

**Review by teleivo — APPROVED** [review]

**Pull request merged**